src/iadmin.cpp

```diff
@@ -2001,10 +2001,13 @@ main( int argc, char **argv ) {
             keepGoing = false;
         }
         if ( firstTime ) {
-            if ( status == 0 ||
-                    status == -2 ) {
+            if (0 == status) {
                 keepGoing = false;
             }
+            if (-2 == status) {
+                keepGoing = false;
+                lastCommandStatus = -2;
+            }
```

Comment:
Please add words explaining why -2 is special. If you know what the distinguishing factor is between the different error codes, put it in a comment as well.

Can even add words to the commit body.